### DISCUSSION TITLE
Version 0.6.1 Beta 2

### DISCUSSION THREAD

**Discussion by DSheirer** [DSheirer/sdrtrunk#2024] [Announcements]
## Features:
* Tuner minimum and maximum frequency now user adjustable so you can lock a tuner to a specific frequency range
* SDRPlay RSP1B & RSPdxR2 tuners and API 3.15 and 3.16 support
* P25 Phase 1 & 2 expanded message decoding, enhanced traffic channel management, talker alias/gps and ISSI/Roam fully qualified Identifiers with alias support
* DMR significant improvements to processing and talker alias/gps support
* Enhanced map display
* Radio Reference API version 18
* Fix for some critical issues causing out-of-memory, application crashes, and channels stuck in TEARDOWN
* Automatic threading dead-lock detection, notification and report creation
* Create diagnostic reports (File >> Reports) for reporting out-of-memory and app crash errors
* Latest Java OpenJDK 23

## Requirements:
* Windows x86 (64-bit)
* Linux x86/aarch64 (64-bit)
* OSX x86/aarch64 (64-bit)

**Note**: Linux for arm-32 is NOT supported

## Optional Requirements
**JMBE version 1.0.9+** is required for decoding APCO25 Phase 1/2 and DMR digital audio on Linux and OSX aarch64 and this library can be compiled within the sdrtrunk application.

Legacy instructions for compiling the JMBE library are located at: **https://github.com/DSheirer/sdrtrunk/wiki/JMBE**

## User Manual
* [Wiki](https://github.com/DSheirer/sdrtrunk/wiki)
* [User Manual](https://github.com/DSheirer/sdrtrunk/wiki/User-Manual)

## Support
* [Support](https://github.com/DSheirer/sdrtrunk/wiki/Support)

## What's Changed
* #1837 Channel Stream Thread Naming by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1838
* #1839 DMR Tone Identifiers Don't Clear At Call End by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1840
* #1841 DMR Call Events Include Channel & Frequency by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1842
* #1843 Map Entity Plotter Concurrent Modification Exception by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1844
* #1845 Tuner Don't Change Frequency Unless Have To by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1846
* #1847 Spectral Display of Multi-Frequency Channel Configurations by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1848
* #172 Tuner Supports Adjusting Min/Max Frequencies by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1850
* #1852 Spectral Display Channel Banner Coloration by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1853
* #1854 Tuner Min Frequency - Allow 1 Hertz Minimum by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1855
* #1856 DMR Channel - Threading Deadlock by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1857
* #1304 P25 TDMA CC Support & Phase 1/2 decoder enhancements. by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1898
* #1899 Radio Reference API V18 & Update Site Editor To Show TDMA CC Sites by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1900
* #1876 Unifies Reentrant Lock Between Tuners and Frequency Controller. by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1901
* #1903 P25 Phase 2 Audio Segment Chopping by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1904
* #1905 P25 Phase 1 Encrypted Call Status by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1906
* #1907 Hytera GPS & XPT Neighbor & Talker Aliases by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1908
* #1849 Map Enhancements by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1920
* #1924 GPS Positions Listed as User Identifiers In Map Panel by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1925
* #1916 SDRPlay API 3.15 and RSPdxR2 support. by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1921
* #1926 Toolkit not initialized error when running headless during APCO… by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1928
* #1931 P25P2 L3Harris Patch Group - Incorrect Radio/Talkgroup Members by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1932
* #1934 P25P1 Conventional Fallback LC Message by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1935
* #1781 Radio Reference API Login Enhanced Logging by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1943
* #1944 SDRPlay API version 3.16 by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1945
* 1860 LibUsb Transfer Buffer Error Handling Enhancements by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1950
* #1952 P25 Traffic Channel Manager Aggregating Multiple Call Events Into One by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1953
* #1912 P25 Patch Group Details Missing In Now Playing and Streaming/Recording Metadata by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1954
* #1955 Java JDK 22 and Gradle 8.10 and Version to beta2 by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1958
* #1960 Application Root Directory  Not Found On Fresh Install by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1961
* #1964 SDRPlay API Native Library on MacOS by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1965
* #1966 P25 Aliases Not Correctly Matching Against Fully Qualified Radio & Talkgroup Values by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1967
* #1948 Audio Streaming Startups On Separate Thread by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1968
* #1974 P25 Unassigned Fully Qualified Talkgroup/Radio Value Handling by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1975
* #1971 P25 AMBTC Status Query - Error Parsing Source Address by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1976
* #1977 P25 FQSUID Improvements by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1978
* #1939 Duplicate Call Detector Enhancements & Unit Tests by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1980
* Improvements to icecast failure handling by @doug-hoffman in https://github.com/DSheirer/sdrtrunk/pull/1401
* #1981 P25P1 Foreign System IDEN_UPD_TDMA Message Support by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1982
* #1983 Revert Aliasing of Fully Qualified Identifiers Against Local ID by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1984
* #1915 (P25) Audio Stops Playing While Calls Continue To Process by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1986
* #1985 Update to JDK 23 by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1987
* #1990 Ignores GPS positions with 0/0 latitude and longitude values. by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1991
* #1994 P25 Phase1 Missing Subsequent Calls Audio When Multiple Conversants On Same Traffic Channel by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1995
* #1996 P25P1 Traffic Channel Monitor Clears Allocated Traffic Channels by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1997
* #1914 P25 Talker Alias Support by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2001
* #2002 P25 Fully Qualified Identifier Formatting For Not Fixed Width Preference by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2003
* #2004 P25P1 Link Control Extended Source Message Processing & ISSI Radio IDs by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2005
* #2007 P25 Call events tracking.  P25 Traffic channel manager now uses… by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2010
* #2006 AudioOutput Recreated Once It Fails To Accept Audio Samples For Playback by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2011
* #2013 Enhance SUID extension message handling and correct encryption … by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2014
* #1973 JUnit Test for Patched Talkgroup Audio Call Streaming by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2015
* #2016 Playback Preferences Null Error by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2017
* #2018 P25 Phase 1 Now Playing Flickering Identifiers by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2019
* #2020 P25 channel null frequency band error. by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2021
* #2022 Audio output reset now reacquires gain and mute controls. by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2023

**Full Changelog**: https://github.com/DSheirer/sdrtrunk/compare/v0.6.1-beta-1...v0.6.1-beta-2

<hr /><em>This discussion was created from the release <a href='https://github.com/DSheirer/sdrtrunk/releases/tag/v0.6.1-beta-2'>Version 0.6.1 Beta 2</a>.</em>

**Comment by gingerhax** [comment]
SOLVED

The below error on launch. I've reinstalled the SDRplay API and JRE

```
Exception in thread "main" java.lang.UnsupportedClassVersionError: io/github/dsheirer/source/tuner/sdrplay/api/v3_07/sdrplay_api_h (class file version 64.65535) was compiled with preview features that are unsupported. This version of the Java Runtime only recognizes preview features for class file version 67.65535
        at java.base/java.lang.ClassLoader.defineClass1(Native Method)
        at java.base/java.lang.ClassLoader.defineClass(Unknown Source)
        at java.base/java.security.SecureClassLoader.defineClass(Unknown Source)
        at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(Unknown Source)
        at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(Unknown Source)
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(Unknown Source)
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(Unknown Source)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(Unknown Source)
        at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
        at io.github.dsheirer.source.tuner.sdrplay.api.device.RspDuoMode.<clinit>(RspDuoMode.java:29)
        at io.github.dsheirer.source.tuner.sdrplay.api.DeviceSelectionMode.<clinit>(DeviceSelectionMode.java:32)
        at io.github.dsheirer.source.tuner.TunerFactory$1.<clinit>(TunerFactory.java:268)
        at io.github.dsheirer.source.tuner.TunerFactory.getUsbTuner(TunerFactory.java:349)
        at io.github.dsheirer.source.tuner.manager.DiscoveredUSBTuner.start(DiscoveredUSBTuner.java:107)
        at io.github.dsheirer.source.tuner.manager.TunerManager.tunerStatusUpdated(TunerManager.java:426)
        at io.github.dsheirer.source.tuner.manager.TunerManager.startAndConfigureTuner(TunerManager.java:317)
        at io.github.dsheirer.source.tuner.manager.TunerManager.addUsbTuner(TunerManager.java:285)
        at io.github.dsheirer.source.tuner.manager.TunerManager.discoverUSBTuners(TunerManager.java:272)
        at io.github.dsheirer.source.tuner.manager.TunerManager.start(TunerManager.java:126)
        at io.github.dsheirer.gui.SDRTrunk.<init>(SDRTrunk.java:184)
        at io.github.dsheirer.gui.SDRTrunk.main(SDRTrunk.java:928)
```

**Reply by DSheirer** [reply]
Sounds like you unzipped onto an existing version and some of the files didn't get replaced.  Try a clean install.

**Reply by gingerhax** [reply]
Solved

**Comment by josephdambrosio** [comment]
Deployed this build to two different sites monitoring two different systems and hearing a ton of digital artifacts and glitching on P25 Phase 2 audio.  Rolling back to beta 1 everything goes back to normal.

**Comment by DSheirer** [comment]
Check the ppm on your tuners.  It may have reset with the latest version
